src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -159,14 +163,19 @@ private List<Location> getPieceLocations(List<Piece> pieces, CompositePoLine poL
       .toList();
   }
 
-  protected Future<List<Piece>> isOperationRestricted(List<Piece> pieces, RequestContext requestContext) {
-    var pieceIds = pieces.stream().map(Piece::getId).toList();
+  private Future<Void> isOperationRestricted(List<String> pieceIds, RequestContext requestContext) {
     return titlesService.getTitlesByPieceIds(pieceIds, requestContext)
       .map(titles -> titles.stream()
         .map(title -> protectionService.isOperationRestricted(title.getAcqUnitIds(), ProtectedOperationType.UPDATE, requestContext))
         .toList())
       .map(GenericCompositeFuture::all)
-      .map(pieces);
+      .mapEmpty();
+  }
+
+  private Future<List<Piece>> validateFetchedPiecesQuantity(List<Piece> pieces, List<String> pieceIds) {
+    return pieces.size() == pieceIds.size()
+      ? Future.succeededFuture(pieces)
+      : Future.failedFuture(new HttpException(HttpStatus.HTTP_BAD_REQUEST.toInt(), ErrorCodes.PIECE_NOT_FOUND));
```

Comment:
better to add which pieces not found or just return all pieces  as a param

Reply:
Good suggestion, included